From 8e9b1c4062c9ef25ee83cc7097506e7469932a22 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 13 Jun 2025 17:34:28 +0300
Subject: [PATCH] [JSW-670] jgs: update address space

Register names are updated as part of jgs specs.
Hence, this patch takes care of those register
renaming.

Change-Id: I550de499e5487e60325e692f82005d7ed2470dd5
---

diff --git a/arch/jgs/arc/palad.c b/arch/jgs/arc/palad.c
index 7a51273..69ee881 100644
--- a/arch/jgs/arc/palad.c
+++ b/arch/jgs/arc/palad.c
@@ -24,6 +24,7 @@
 #include <jgs_efuse.h>
 #include <efuse.h>
 #include <hl.h>
+#include <soc.h>
 #include <address_space.h>
 #include <lib_common.h>
 #include "auto_gen_pcie_phy_csr_locations.h"
@@ -76,18 +77,18 @@
 #if defined(DIE_TYPE_IO)
 void hl_uart_init(void)
 {
-	regmap_writel(1, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE +
+	regmap_writel(1, mmIO_0_CFG_PSOC_PSOC_CFG_PAD_CONF_BASE +
 			mmPSOC_PAD_CONF_PAD_SEL_49);	/* UART0 RXD */
-	regmap_writel(1, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE +
+	regmap_writel(1, mmIO_0_CFG_PSOC_PSOC_CFG_PAD_CONF_BASE +
 			mmPSOC_PAD_CONF_PAD_SEL_50);	/* UART0 TXD */
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PALAD_UART_BAUDRATE);
 }
 #else
 void hl_uart_init(void)
 {
-	regmap_writel(1, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE +
+	regmap_writel(1, mmIO_0_CFG_PSOC_PSOC_CFG_PAD_CONF_BASE +
 			mmPSOC_PAD_CONF_PAD_SEL_33);	/* UART0 RXD */
-	regmap_writel(1, mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE +
+	regmap_writel(1, mmIO_0_CFG_PSOC_PSOC_CFG_PAD_CONF_BASE +
 			mmPSOC_PAD_CONF_PAD_SEL_34);	/* UART0 TXD */
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PALAD_UART_BAUDRATE);
 }
@@ -131,7 +132,7 @@
 	u8 num_vf = 0;
 	u32 val;
 
-	hl_fuse_read(mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_EFUSE_BASE, IO_EFUSE_PALAD_CFG_0_OFFSET,
+	hl_fuse_read(mmIO_0_CFG_PSOC_PSOC_CFG_EFUSE_BASE, IO_EFUSE_PALAD_CFG_0_OFFSET,
 		     &pci_efuse_cfg);
 	preboot_puts_hex("efuse pci cfg = ", pci_efuse_cfg, false);
 
@@ -190,40 +191,40 @@
 	 * The exact size of data written to SRAM_LOCATIONS and CSR_LOCATIONS
 	 * is written to PHY_FW_FSM_SIZE register.
 	 */
-	val = regmap_readl(mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	val = regmap_readl(mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_SIZE);
 	val &= ~PCIE_FW_FSM_PHY_FW_FSM_SIZE_PMA_SRAM_SIZE_M;
 	val |= PHY_FW_SIZE << PCIE_FW_FSM_PHY_FW_FSM_SIZE_PMA_SRAM_SIZE_S;
-	regmap_writel(val, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	regmap_writel(val, mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_SIZE);
 
-	val = regmap_readl(mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	val = regmap_readl(mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_SIZE);
 	val &= ~PCIE_FW_FSM_PHY_FW_FSM_SIZE_PMA_CSR_SIZE_M;
 	val |= (uint16_t)get_csr_locations_data_size() << PCIE_FW_FSM_PHY_FW_FSM_SIZE_PMA_CSR_SIZE_S;
-	regmap_writel(val, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	regmap_writel(val, mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_SIZE);
 
 	/*
 	 * PHY_FW_FSM_CTRL ➔ SRAM_ROM_SEL should be set to '0'
 	 * to disable default phy fw from ROM.
 	 */
-	val = regmap_readl(mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	val = regmap_readl(mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_CTRL);
 	val &= ~PCIE_FW_FSM_PHY_FW_FSM_CTRL_SRAM_ROM_SEL_M;
 	val |= 0 << PCIE_FW_FSM_PHY_FW_FSM_CTRL_SRAM_ROM_SEL_S;
-	regmap_writel(val, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	regmap_writel(val, mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_CTRL);
 
 	/*
 	 * PHY_FW_FSM_CTRL ➔ FSM_START set to '1'
 	 * to start the phy fw loading to the fw memories.
 	 */
-	val = regmap_readl(mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	val = regmap_readl(mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_CTRL);
 	val &= ~PCIE_FW_FSM_PHY_FW_FSM_CTRL_FSM_START_M;
 	val |= 1 << PCIE_FW_FSM_PHY_FW_FSM_CTRL_FSM_START_S;
-	regmap_writel(val, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	regmap_writel(val, mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 			   mmPCIE_FW_FSM_PHY_FW_FSM_CTRL);
 
 	/*
@@ -237,7 +238,7 @@
 	 * once pcie phy is implemented, this code should be uncommented.
 	 * jira ticket: H15-352
 	 *
-	 * while (!(regmap_readl(mmIO_0_CFG_HALF_0_CXL_EP_PCIE_FW_FSM_BASE +
+	 * while (!(regmap_readl(mmIO_0_CFG_CXL_EP_PCIE_FW_FSM_BASE +
 	 * mmPCIE_FW_FSM_PHY_FW_FSM_CTRL) &
 	 * PCIE_FW_FSM_PHY_FW_FSM_CTRL_FW_FSM_PHY_INIT_DONE_M));
 	 *
@@ -294,7 +295,7 @@
 	regmap_writel(val, IO0_H0_PCIE_DBI(mmPCIE_DBI_MISC_CONTROL_1_OFF));
 
 	/* switch to shadow registers access in order to set BAR masks */
-	regmap_writel(PCIE_AUX_DBI_CS2_M, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_AUX_BASE + mmPCIE_AUX_DBI);
+	regmap_writel(PCIE_AUX_DBI_CS2_M, mmIO_0_CFG_CXL_EP_PCIE_AUX_BASE + mmPCIE_AUX_DBI);
 
 	/* set PF CFG BAR to 1GB */
 	regmap_writel(BAR_SIZE_1GB, IO0_H0_PCIE_DBI(mmPCIE_DBI_BAR0_REG));
@@ -305,7 +306,7 @@
 	regmap_writel(0, IO0_H0_PCIE_DBI(mmPCIE_DBI_SRIOV_BAR1_REG));
 
 	/* disable shadow registers access */
-	regmap_writel(0, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_AUX_BASE + mmPCIE_AUX_DBI);
+	regmap_writel(0, mmIO_0_CFG_CXL_EP_PCIE_AUX_BASE + mmPCIE_AUX_DBI);
 
 	/* set VF resizable BAR #2 allowed size */
 	regmap_writel(VF_RES_BAR_SIZES, IO0_H0_PCIE_DBI(mmPCIE_DBI_VF_RESBAR_CAP_REG_2_REG));
@@ -326,7 +327,7 @@
 
 	palad_pcie_params_init();
 
-	regmap_writel(LTTSM_ENA_MASK, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_AUX_BASE +
+	regmap_writel(LTTSM_ENA_MASK, mmIO_0_CFG_CXL_EP_PCIE_AUX_BASE +
 				      mmPCIE_AUX_LTSSM_EN);
 
 	/* release request retry so the host can access pci config space */
@@ -377,7 +378,7 @@
 	 * (or in RST ICCM for COMP-DIE).
 	 * An efuse bit is controlling whether the application binary is larger than available.
 	 */
-	hw_mem_copy(IO0_MISC_ICCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL, IO_MISC_ICCM_SIZE, false, false);
+	hw_mem_copy(IO_0_MISC_ARC_ICCM_BASE_FULL, PREBOOT_TEXT_BASE_FULL, IO_0_MISC_ARC_ICCM_SIZE, false, false);
 
 	err = preboot_fuse_read(IO_EFUSE_PALAD_CFG_0_OFFSET, &val);
 	if (err)
@@ -388,8 +389,8 @@
 	if (!val)
 		return;
 
-	hw_mem_copy(IO0_MISC_DCCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL + IO_MISC_ICCM_SIZE,
-			IO_MISC_DCCM_SIZE, false, false);
+	hw_mem_copy(IO_0_MISC_ARC_DCCM_BASE_FULL, PREBOOT_TEXT_BASE_FULL + IO_0_MISC_ARC_ICCM_SIZE,
+			IO_0_MISC_ARC_DCCM_SIZE, false, false);
 }
 #else
 static void btldr_copy_preboot_app(void)
